src/utils/createRequirePropertyRule.ts

```diff
@@ -0,0 +1,42 @@
+import { createRule } from "../createRule.js";
+
+/**
+ * Given a top-level property name, create a rule that requires that property to be present.
+ * Optionally, include it in the recommended config.
+ */
+export const createRequirePropertyRule = (
+	propertyName: string,
+	isRecommended = false,
+) => {
+	return createRule({
+		create(context) {
+			let hasSeen = false;
```

Comment:
[Refactor] This _"wait until the end"_ strategy adds a bit more complexity than we need. Since we know the property must be a child of the object expression, we can directly check its properties:

```ts
"Program > JSONExpressionStatement > JSONObjectExpression"(
	node: JsonAST.JSONObjectExpression,
) {
	if (
		!node.properties.some(
			(property) =>
				isJSONStringLiteral(property.key) &&
				property.key.value === propertyName,
		)
	) {
		context.report({
			data: { property: propertyName },
			messageId: "missing",
			node: context.sourceCode.ast,
		});
	}
},
```